src/bin/main.rs

```diff
@@ -50,31 +50,44 @@ fn run(args: &ArgMatches) -> HowserResult<()> {
         }
         ("validate", Some(sub_m)) => {
             let options = vec![CLIOption::VerboseMode(sub_m.is_present("verbose"))];
-            if sub_m.is_present("pharmacy") {
-                let fail_early = sub_m.is_present("fail-early");
-                let filename = args
-                    .value_of("pharmacy").
-                    ok_or(HowserError::RuntimeError(
-                        "Pharmacy filename could not be parsed from the argument string.".to_string()))?;
-                let pharmacy_contents = get_file_contents(filename)?;
-                let specs = &pharmacy_contents.parse::<Value>()?["Specs"];
-                let prescription_pairs = specs
-                    .as_table()
-                    .ok_or(HowserError::RuntimeError(
-                        format!("Error parsing pharmacy file {}.", filename)))?;
-
-                (process_pharmacy_file(prescription_pairs, fail_early)?, options)
-            } else {
-                let rx_name = sub_m
-                    .value_of("prescription")
-                    .ok_or(HowserError::RuntimeError(
-                        "Unable to parse the name of the prescription file.".to_string()))?;
-                let document_name = sub_m
-                    .value_of("document")
-                    .ok_or(HowserError::RuntimeError(
-                        "Unable to parse the name of the document file.".to_string()))?;
-
-                (validate(rx_name, document_name)?, options)
+            let rx_name = sub_m
+                .value_of("prescription")
+                .ok_or(HowserError::RuntimeError(
+                    "Unable to parse the name of the prescription file.".to_string()))?;
+            let document_name = sub_m
+                .value_of("document")
+                .ok_or(HowserError::RuntimeError(
+                    "Unable to parse the name of the document file.".to_string()))?;
+
+            (validate(rx_name, document_name)?, options)
+        },
+        ("pharmacy", Some(sub_m)) => {
+            match sub_m.subcommand() {
+                ("check", Some(sub_m)) => {
+                    let options = vec![CLIOption::VerboseMode(sub_m.is_present("verbose"))];
+                    let fail_early = sub_m.is_present("fail-early");
+                    let filename = sub_m
+                        .value_of("pharmacy").
+                        ok_or(HowserError::RuntimeError(
+                            "Pharmacy filename could not be parsed from the argument string.".to_string()))?;
+                    let pharmacy = parse_pharmacy_file(filename)?;
+                    let keys: Vec<_> = pharmacy.keys().collect();
```

Comment:
If you want to extend the notion of `pharmacy check` just a little bit further to parity with the single spec check variant, it would probably be to check the correctness of the pharmacy.toml file itself. E.G. confirming that all the key-value pairs have strings for their values.

This is a completeness suggestion, and not a blocking concern.